orchagent/orchdaemon.cpp

```diff
@@ -124,11 +124,54 @@ bool OrchDaemon::init()
             SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES,
         };
 
+        static const vector<sai_queue_attr_t> queueAttrIds;
+
         m_orchList.push_back(new PfcWdSwOrch<PfcWdZeroBufferHandler, PfcWdLossyHandler>(
                     m_applDb,
                     pfc_wd_tables,
                     portStatIds,
-                    queueStatIds));
+                    queueStatIds,
+                    queueAttrIds));
+    }
+    else
+    {
+        static const vector<sai_port_stat_t> portStatIds =
+        {
+            SAI_PORT_STAT_PFC_0_RX_PKTS,
+            SAI_PORT_STAT_PFC_1_RX_PKTS,
+            SAI_PORT_STAT_PFC_2_RX_PKTS,
+            SAI_PORT_STAT_PFC_3_RX_PKTS,
+            SAI_PORT_STAT_PFC_4_RX_PKTS,
+            SAI_PORT_STAT_PFC_5_RX_PKTS,
+            SAI_PORT_STAT_PFC_6_RX_PKTS,
+            SAI_PORT_STAT_PFC_7_RX_PKTS,
+            SAI_PORT_STAT_PFC_0_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_1_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_2_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_3_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_4_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_5_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_6_ON2OFF_RX_PKTS,
+            SAI_PORT_STAT_PFC_7_ON2OFF_RX_PKTS,
+        };
+
+        static const vector<sai_queue_stat_t> queueStatIds =
+        {
+            SAI_QUEUE_STAT_PACKETS,
+            SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES,
+        };
+
+        static const vector<sai_queue_attr_t> queueAttrIds =
+        {
+            SAI_QUEUE_ATTR_PAUSE_STATUS,
+        };
+
+        m_orchList.push_back(new PfcWdSwOrch<PfcWdActionHandler, PfcWdActionHandler>(
```

Comment:
I think you can reuse forward handler

Reply:
Currently both drop and forward uses base class for simplicity. I agree we might reuse forward handler. This PR is for alert mode. So will update drop/forward handler together in a separate PR.

Reply:
ok